src/components/ShoppingList/ShoppingList.js

```diff
@@ -29,13 +29,28 @@ const ShoppingList = () => {
     setItems(filteredItems);
   };
 
+  const updateItem = (edit, key) => {
+    const updatedItems = items.map((item) => {
+      const newItem = item;
+      if (item.key === key) {
+        newItem.text = edit;
```

Comment:
It's best to not mutate the state directly and just return the new value, i.e. change this line to:
```suggestion
        return {
          ...item,
          text: edit,
        };
```